CMakeLists.txt

```diff
@@ -57,8 +57,10 @@ target_link_libraries(ghex_common INTERFACE Boost::boost)
 target_link_libraries(ghex_common INTERFACE GridTools::gridtools)
 target_link_libraries(ghex_common INTERFACE oomph::oomph)
 target_link_libraries(ghex PRIVATE MPI::MPI_CXX)
+target_link_libraries(ghex PRIVATE oomph::oomph_mpi)
```

Comment:
This seems wrong to me, should be `target_link_libraries(ghex INTERFACE oomph::oomph)`

Reply:
I agree, this line is wrong there.. the function `ghex_link_to_oomph` will choose the specialized library to link.
However, without any change in cmake files I get a link failure:
```
[9/34] Linking CXX shared library lib/libghex.dylib
      FAILED: lib/libghex.dylib
      : && /Library/Developer/CommandLineTools/usr/bin/c++ -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk -dynamiclib -Wl,-headerpad_max_install_names -L/opt/homebrew/Cellar/libevent/2.1.12_1/lib -o lib/libghex.dylib -install_name @rpath/libghex.dylib CMakeFiles/ghex.dir/src/context.cpp.o  -Wl,-rpath,/var/folders/5x/0ckg0y5n76z6gqk8_cxdzbgh0000gn/T/tmplh36jpu6/build/lib  /opt/homebrew/Cellar/open-mpi/4.1.5/lib/libmpi.dylib  lib/libhwmalloc.dylib && :
      ld: Undefined symbols:
        oomph::util::heap_pimpl<oomph::detail::message_buffer::heap_ptr_impl>::heap_pimpl(oomph::util::heap_pimpl<oomph::detail::message_buffer::heap_ptr_impl>&&), referenced from:
            ghex::context::make_buffer(unsigned long) in context.cpp.o
        oomph::detail::message_buffer::~message_buffer(), referenced from:
            ghex::context::make_buffer(unsigned long) in context.cpp.o
        oomph::context::make_buffer_core(unsigned long), referenced from:
            ghex::context::make_buffer(unsigned long) in context.cpp.o
        oomph::context::context(ompi_communicator_t*, bool, bool, unsigned long), referenced from:
            ghex::context::context(ompi_communicator_t*, bool) in context.cpp.o
            ghex::context::context(ompi_communicator_t*, bool) in context.cpp.o
        oomph::context::~context(), referenced from:
            std::__1::unique_ptr<oomph::context, std::__1::default_delete<oomph::context>>::~unique_ptr[abi:v160006]() in context.cpp.o
            ghex::context::context(ompi_communicator_t*, bool) in context.cpp.o
```

I have pushed a new commit with another solution which works for me, but please suggest if you have a better one.